City-of-Helsinki/drupal-hdbt
# UHF-5527 Small accessibility improvements for unit searches and custom searches

# [UHF-5527](https://helsinkisolutionoffice.atlassian.net/browse/UHF-5527)


## What was done


* Added role="search" to form element in unit searches and custom searches.
* Added type="search" to inputs that were previously type text in forms in unit searches and custom searches.
* Removed magnifying glass icon from the search inputs in unit searches and custom searches.

## How to install

* Make sure your instance is up and running on latest dev branch. I recommend testing this with all instances that are listed below in the testing instructions. 
* For KASKO and STRATEGIA there is feature-branch that needs to be used as well.
  * `git pull origin dev`
  * `make fresh`
OR
  * `git fetch`
  * `git checkout UHF-5527`
  * `make fresh`
* Update the HDBT theme
  * `composer require drupal/hdbt:dev-UHF-5527`
* Run `make drush-cim drush-cr`

## How to test


Make sure that for each of these listed searches the form element has attribute role="search" and that if there is text-input field on the filters it has the type="search" attribute. Also the input field should no longer have the magnifying glass icon on any of the searches. ATTENTION: the ploughing schedule search uses HDS-SearchInput component and that component doesn't support type-attribute so it doesn't have it.
### KASKO:
* [x] https://helfi-kasko.docker.so/fi/kasvatus-ja-koulutus/perusopetus/peruskoulut
* [x] https://helfi-kasko.docker.so/fi/kasvatus-ja-koulutus/varhaiskasvatus/varhaiskasvatus-paivakodissa/etsi-kunnallisia-paivakoteja
* [x] https://helfi-kasko.docker.so/fi/kasvatus-ja-koulutus/lukiokoulutus/lukiot
* [x] https://helfi-kasko.docker.so/fi/kasvatus-ja-koulutus/leikkipuistot/leikkipuistot-ja-perhetalot
* [x] https://helfi-kasko.docker.so/fi/kasvatus-ja-koulutus/ammatillinen-koulutus/ammatilliset-oppilaitokset
### STRATEGIA:
* [x] https://strategia.docker.so/fi/paatoksenteko-ja-hallinto/ota-yhteytta-helsingin-kaupunkiin/neuvontapalvelut
### SOTE:
* [x] https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/senioripalvelut/kotihoito
* [x] https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/senioripalvelut/palvelukeskukset
* [x] https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/senioripalvelut/seniorikeskukset-etusivu
* [x] https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/senioripalvelut/ymparivuorokautinen-asuminen-ja-hoito
* [x] https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/terveydenhoito/hammashoito/hammashoitolat
* [x] https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/terveydenhoito/kuntoutus-ja-terapiat/toimintaterapia
* [x] https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/terveydenhoito/terveysasemat
* [x] https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/vammaispalvelut/asuminen
* [x] https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/vammaispalvelut/asuminen/kotihoito
* [x] https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/vammaispalvelut/tyo-ja-paivatoiminta
* [x] https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/terveyskeskushaku
* [x] https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/neuvolahaku
### KUVA:
* [x] https://helfi-kuva.docker.so/fi/kulttuuri-ja-vapaa-aika/ulkoilu-puistot-ja-luontokohteet/uimarannat/helsingin-uimarannat
### REKRY:
* [x] https://helfi-rekry.docker.so/fi/avoimet-tyopaikat/avoimet-tyopaikat
### KYMP:
* [ ] https://helfi-kymp.docker.so/fi/kaupunkiymparisto-ja-liikenne/kaupunkisuunnittelu-ja-rakentaminen/suunnitelmat-ja-rakennushankkeet
* [x] https://helfi-kymp.docker.so/fi/kaupunkiymparisto-ja-liikenne/kunnossapito/katujen-kunnossapito/katujen-talvikunnossapito
### ETUSIVU:
* [x] https://helfi-etusivu.docker.so/en/news


## Designers review



## Other PRs


* https://github.com/City-of-Helsinki/drupal-hdbt/pull/902
* https://github.com/City-of-Helsinki/drupal-helfi-kasvatus-koulutus/pull/566
* https://github.com/City-of-Helsinki/drupal-helfi-strategia/pull/578


[UHF-5527]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-5527?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
